imas_muscle3/actors/visualization_component.py

```diff
@@ -52,64 +58,68 @@ def main() -> None:
     last_trigger_time = 0.0
+    ports_in = [
+        p
+        for p in get_port_list(instance, Operator.S)
+        if not p.endswith("_md_in")
+    ]
     while instance.reuse_instance():
-        if first_run:
-            plot_file_path = instance.get_setting("plot_file_path", "str")
-            # If port is not specified, use a random available port
-            port = get_setting_optional(instance, "port", 0)
-            # FIXME: there is an issue when the plotting takes much longer
-            # than it takes for data to arrive from the MUSCLE actor. As a
-            # remedy, set a plotting throttle interval.
-            throttle_interval = get_setting_optional(
-                instance, "throttle_interval", 0.1
-            )
-            keep_alive = get_setting_optional(instance, "keep_alive", False)
-            open_browser = get_setting_optional(instance, "open_browser", True)
-            automatic_mode = get_setting_optional(
-                instance, "automatic_mode", False
-            )
-            extract_all = get_setting_optional(
-                instance, "automatic_extract_all", False
-            )
-
-            md_dict = handle_machine_description(instance)
-
-            assert (
-                port is not None
-                and open_browser is not None
-                and extract_all is not None
-                and automatic_mode is not None
-            )
-            visualization_actor = VisualizationActor(
-                plot_file_path,
-                port,
-                md_dict,
-                open_browser,
-                extract_all,
-                automatic_mode,
-            )
-            first_run = False
+        if instance.resuming():
+            pass
+        if instance.should_init():
+            pass
+
+        plot_file_path = instance.get_setting("plot_file_path", "str")
+        # If port is not specified, use a random available port
+        port = get_setting_optional(instance, "port", 0)
+        # FIXME: there is an issue when the plotting takes much longer
+        # than it takes for data to arrive from the MUSCLE actor. As a
+        # remedy, set a plotting throttle interval.
+        throttle_interval = get_setting_optional(
+            instance, "throttle_interval", 0.1
+        )
+        keep_alive = get_setting_optional(instance, "keep_alive", False)
+        open_browser = get_setting_optional(instance, "open_browser", True)
+        automatic_mode = get_setting_optional(
+            instance, "automatic_mode", False
+        )
+        extract_all = get_setting_optional(
+            instance, "automatic_extract_all", False
+        )
+
         assert (
-            visualization_actor is not None and throttle_interval is not None
+            port is not None
+            and open_browser is not None
+            and extract_all is not None
+            and automatic_mode is not None
+            and throttle_interval is not None
         )
```

Comment:
Isn't it easier to have separate assert statements here? Then you'll even know which of these is the problem ;)

Also beware that assert statements may be disabled when running python with the `-O` flag, or `PYTHON_OPTIMIZE=1` environment variable. If you need to be sure that these conditions are adhered to, you should replace it with `if not ...: raise RuntimeError(...)`

See also: https://docs.python.org/3/using/cmdline.html#cmdoption-O